applications/ConstitutiveLawsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.cpp

```diff
@@ -77,59 +78,59 @@ void GenericSmallStrainHighCycleFatigueLaw<TConstLawIntegratorType>::CalculateMa
         double uniaxial_stress;
         TConstLawIntegratorType::YieldSurfaceType::CalculateEquivalentStress(predictive_stress_vector, r_strain_vector, uniaxial_stress, rValues);
 
-        double max_stress = this->GetMaxStress();
-        double min_stress = this->GetMinStress();
-        double sign_factor = HighCycleFatigueLawIntegrator<6>::CalculateTensionCompressionFactor(predictive_stress_vector);
-        uniaxial_stress *= sign_factor;
-        unsigned int global_number_of_cycles = this->GetNumberOfCyclesGlobal();
-        unsigned int local_number_of_cycles = this->GetNumberOfCyclesLocal();
-        bool max_indicator = this->GetMaxDetected();
-        bool min_indicator = this->GetMinDetected();
-        double fatigue_reduction_factor = this->GetFatigueReductionFactor();
-        double B0 = this->GetFatigueReductionParameter();
-        double previous_max_stress = this->GetPreviousMaxStress();
-        double previous_min_stress = this->GetPreviousMinStress();
-        double wohler_stress = this->GetWohlerStress();
-
+        double max_stress = mMaxStress;
```

Comment:
Do you need to copy all that values (are they constant)

Reply:
You are right; this could be improved in several ways. I need to carefully think about this because some of them are needed. Despite this, the CL works properly at this moment and the changes proposed in this PR really enhanced the current version of the CL, so before changing things I would like to merge this version and then deal have a closer look to this.